pkg/controller/status.go

```diff
@@ -167,6 +169,24 @@ func (c *Controller) GetWorkersCnt() uint32 {
 	return c.opConfig.Workers
 }
 
+//WorkerStatus provides status of the worker
+func (c *Controller) WorkerStatus(workerID uint32) (*spec.WorkerStatus, error) {
+	obj, ok := c.curWorkerCluster.Load(workerID)
+	if !ok || obj == nil {
+		return nil, fmt.Errorf("worker has no status")
+	}
+
+	cl, ok := obj.(*cluster.Cluster)
+	if !ok {
+		return nil, fmt.Errorf("could not cast to Cluster struct")
+	}
+
+	return &spec.WorkerStatus{
+		CurrentCluster: util.NameFromMeta(cl.ObjectMeta),
```

Comment:
I think it's not concurrency safe: the properties and methods of the cluster are accessed here without any locks, neither locks are taken when they are modified. That may lead to the cluster.currentProcess being accessed while it is modified. I'd make a  special mutex and lock it on the setProcessName and GetCurrentProcess (and also the place where you change the start time).

Reply:
Isn't it safe but maybe outdated?